src/lib/fastAuthWalletConnection.ts

```diff
@@ -504,4 +519,98 @@ export class FastAuthWalletConnection {
       window.addEventListener('message', listener);
     });
   }
+
+  async requestSignMultiChain(data: SendMultichainMessage) {
+    return this._requestToIframe({
+      evtTypes: {
+        loaded: 'signMultiChainLoaded',
+        request: 'multiChainRequest',
+        response: 'multiChainResponse',
+      },
+      data,
+      urlPath: '/sign-multichain/',
+    });
+  }
+
+  async requestSignMessage(
+    data: SignMessageParams
+  ): Promise<SignedMessage | void> {
+    const res = (
+      await this._requestToIframe<
+        SignMessageParams,
+        { data: { ok: boolean; data: SignedMessage } }
+      >({
+        evtTypes: {
+          loaded: 'signMessageLoaded',
+          request: 'signMessageRequest',
+          response: 'signMessageResponse',
+        },
+        data,
+        urlPath: '/sign-message/',
+      })
+    ).data;
+
+    if (!res.ok) {
+      throw new Error('Failed to sign message');
+    }
+
+    return res.data;
+  }
+
+  async verifySignMessage(
+    message: SignMessageParams,
+    signedMessage: SignedMessage
+  ): Promise<boolean> {
+    const signatureBytes = Uint8Array.from(
+      Buffer.from(signedMessage.signature, 'base64')
+    );
+    const keyPair = PublicKey.fromString(signedMessage.publicKey);
+
+    class Payload {
+      tag: number;
+      message: string;
+      nonce: number[];
+      recipient: string;
+      callbackUrl: string;
+
+      constructor({ message, nonce, recipient, callbackUrl }: Payload) {
+        this.tag = 2147484061;
+        Object.assign(this, {
+          message,
+          nonce,
+          recipient,
+          callbackUrl,
+        });
+      }
+    }
+
+    const payload = new Payload({
+      tag: 2147484061,
```

Comment:
if tag value will be consistent, can we pull it out to a constant? (saw it being used two places)

Reply:
Good point, I will setup the tag in a single place. It's idd duplicated